LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js

```diff
@@ -274,11 +279,69 @@ window.test_driver_internal.action_sequence = async function(sources)
         case "pointer":
             pointerSource = source;
             break;
+        case "key":
+            keySource = source;
+            break;
         default:
             throw new Error(`Unknown source type "${source.type}".`);
         }
     }
 
+    if (keySource && pointerSource)
+        throw new Error("testdriver-vendor.js for WebKit does not yet support mixing key and pointer sources");
+
+    if (keySource) {
+        let modifiersInEffect = [];
+        const events = [];
+        for (const action of keySource.actions) {
+            switch (action.type) {
+            case 'keyDown': {
+                const key = convertSeleniumKeyCode(action.value);
+                if (key.modifier)
+                    modifiersInEffect.push(key.modifier);
+                events.push({type: 'rawKeyDown', arguments: [key.key, modifiersInEffect.slice(0)]});
+                break;
+            }
+            case 'keyUp': {
+                const key = convertSeleniumKeyCode(action.value);
+                if (key.modifier)
+                    modifiersInEffect = modifiersInEffect.filter((modifier) => modifier != key.modifier);
+                events.push({type: 'rawKeyUp', arguments: [key.key, modifiersInEffect.slice(0)]});
+                break;
+            }
+            case 'pause':
+                // FIXME: Use eventSender.leapForward.
+                throw new Error('testdriver-vendor.js for WebKit does not yet support pause key action');
```

Comment:
Can we use the `pause` function (basically a setTimeout wrapped inside a Promise) ?

Reply:
So as far as I can tell, there is no imported WPT which uses this pause function so I left it unimplemented.